src/wp-admin/includes/image.php

```diff
@@ -647,18 +647,27 @@ function wp_generate_attachment_metadata( $attachment_id, $file ) {
  * @since 2.5.0
  *
  * @param string $str
- * @return int|float
+ * @return int|float Returns originally passed string on failure.
  */
 function wp_exif_frac2dec( $str ) {
-	if ( false === strpos( $str, '/' ) ) {
+	// Fractions must contain a single `/`.
+	if ( substr_count( $str, '/' ) !== 1 ) {
```

Comment:
What about if someone passed a non-string ? Would it be an idea to use still slightly more defensive coding ?
```suggestion
	if ( ! is_scalar( $str ) || substr_count( $str, '/' ) !== 1 ) {
```

Suggestion: let's also add a couple of test cases with non-scalar input, like `null` and `array( '22/4' )`.
These would fail with a PHP error without the above adjustment.
(`null` would lead to a "passing null to non-nullable" deprecation notice in PHP 8.1 and `array()` would lead to a warning in PHP 7.x and a fatal error in PHP 8.x)

Reply:
great suggestion, will do!